File: kotlin/grade-school/src/main/kotlin/School.kt
class School {
    private var db: Map<Int, List<String>> = emptyMap()

    fun db() = db

    fun add(studentName: String, studentGrade: Int) {
        db += studentGrade to (grade(studentGrade) + studentName)
    }

    fun grade(grade: Int): List<String> {
        return db.get(grade).orEmpty()
    }

    fun sort(): Map<Int, List<String>> {
        return db.toSortedMap().mapValues { it.value.sorted() }
    }
}

File: scala/grade-school/src/main/scala/School.scala
class School {
  type DB = Map[Int, Seq[String]]
  var data: DB = Map.empty

  def add(name: String, g: Int) = {
    data = data ++ Map(g -> (data.getOrElse(g, Seq.empty) ++ Seq(name)))
  }

  def db: DB = data

  def grade(g: Int): Seq[String] = ???

  def sorted: DB = ???
}
